channeldb/graph.go

```diff
@@ -872,6 +863,378 @@ func (c *ChannelGraph) ChannelID(chanPoint *wire.OutPoint) (uint64, error) {
 	return chanID, nil
 }
 
+// TODO(roasbeef): allow updates to use Batch?
+
+// HighestChanID returns the "highest" known channel ID in the channel graph.
+// This represents the "newest" channel from the PoV of the chain. This method
+// can be used by peers to quickly determine if they're graphs are in sync.
+func (c *ChannelGraph) HighestChanID() (uint64, error) {
+	var cid uint64
+
+	err := c.db.View(func(tx *bolt.Tx) error {
+		edges := tx.Bucket(edgeBucket)
+		if edges == nil {
+			return ErrGraphNoEdgesFound
+		}
+		edgeIndex := edges.Bucket(edgeIndexBucket)
+		if edgeIndex == nil {
+			return ErrGraphNoEdgesFound
+		}
+
+		// In order to find the highest chan ID, we'll fetch a cursor
+		// and use that to seek to the "end" of our known rage.
+		cidCursor := edgeIndex.Cursor()
+
+		lastChanID, _ := cidCursor.Last()
+
+		// If there's no key, then this means that we don't actually
+		// know of any channels, so we'll return a predicable error.
+		if lastChanID == nil {
+			return ErrGraphNoEdgesFound
+		}
+
+		// Otherwise, we'll de serialize the channel ID and return it
+		// to the caller.
+		cid = byteOrder.Uint64(lastChanID)
+		return nil
+	})
+	if err != nil && err != ErrGraphNoEdgesFound {
+		return 0, err
+	}
+
+	return cid, nil
+}
+
+// ChannelEdge represents the complete set of information for a channel edge in
+// the known channel graph. This struct couples the core information of the
+// edge as well as each of the known advertised edge policies.
+type ChannelEdge struct {
+	// Info contains all the static information describing the channel.
+	Info *ChannelEdgeInfo
+
+	// Policy1 points to the "first" edge policy of the channel containing
+	// the dynamic information required to properly route through the edge.
+	Policy1 *ChannelEdgePolicy
+
+	// Policy2 points to the "second" edge policy of the channel containing
+	// the dynamic information required to properly route through the edge.
+	Policy2 *ChannelEdgePolicy
+}
+
+// ChanUpdatesInHorizon returns all the known channel edges which have at least
+// one edge that has an update timestamp within the specified horizon.
+func (c *ChannelGraph) ChanUpdatesInHorizon(startTime, endTime time.Time) ([]ChannelEdge, error) {
+	var edgesInHorizon []ChannelEdge
+
+	err := c.db.View(func(tx *bolt.Tx) error {
```

Comment:
I'd say yes, if the processing function were used in other locations in the package/file. Atm, they aren't. The general code style in the project is to favor a clear control flow over excessive function modularization. In this case, you can read a single function and grok the logic rather than following around several other newly introduced functions. 

Not sure what screen you use for viewing code, but it fits on mine ;)